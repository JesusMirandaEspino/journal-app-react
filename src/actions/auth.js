import { types } from '../types/types';

export const login = ( uid, displaName ) => (    {
        type: types.login,
        payload: {
            uid,
            displaName
    }
});
