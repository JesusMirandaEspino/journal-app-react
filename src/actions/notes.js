import { db } from '../firebase/firebaseConfig';
import { collection, addDoc, doc, updateDoc } from 'firebase/firestore';
import { types } from '../types/types';
import { loadNotes } from '../helpers/loadNote';


export const startNewNote = () => {
    return async ( dispatch, getState ) => {
        const { uid } = getState().auth;

        const newNote = {
            title: '',
            body: '',
            date: new Date().getTime()
        }

        const doc = await addDoc(collection(db, `${ uid }`, "journal/notes"), newNote );

        dispatch(  activeNote( doc.id, newNote ) );

    };
}

export const activeNote = ( id, note ) => ({
    type: types.notesActive,
    payload: {
        id,
        ...note
    }
});


export const setNotes = ( notes ) => ({
    type: types.notesLoad,
    payload: notes
});

export const startLoadNotes = ( uid ) => {
    return async ( dispatch ) => {
            const notes = await loadNotes(  uid );
            dispatch(  setNotes(notes) );
    }
}

export const startSaveNotes = ( note ) => {

    return async( dispatch, getState ) => {
        const { uid } = getState().auth;

        if ( !note.url ) {
            delete note.url
        }

        const noteToFirestore = { ...note };
        delete noteToFirestore.id

        const noteRef = doc(db, `${uid}/journal/notes/${note.id}`)
        await updateDoc(noteRef,noteToFirestore);

        dispatch( refreshNote( note.id, noteToFirestore ) );

    }
}


export const refreshNote = ( id, note ) => ({
    type: types.notesUpdate,
    payload: {
        id,
        note: {
            id,
            ...note
        }
    }
});

