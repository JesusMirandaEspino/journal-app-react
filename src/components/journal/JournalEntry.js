import React from 'react'

export const JournalEntry = () => {
    return (
        <div className="journal__entry pointer">

            <div className="journal__entry-picture"  style={ { backgroundSize: 'cover', backgroundImage:'url(https://midu.dev/images/wallpapers/javascript-grande-horizontal-4k.png)' } }>

            </div>

            <div className="journal__entry-body" >

                <p className="journal__entry-title" > Nuevo entry </p>
                <p className="journal__entry-content" > -**************************    </p>

            </div>


            <div className="journal__entry-date-box" >
                <span>Monday</span>
                <h4>28</h4>
            </div>
            
        </div>
    )
}
