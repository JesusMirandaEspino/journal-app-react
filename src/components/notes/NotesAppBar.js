import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { startSaveNotes } from '../../actions/notes';

export const NotesAppBar = () => {

    const dispatch = useDispatch( );
    const { active } = useSelector(  state => state.notes );

    const handleSave = () => {
        dispatch( startSaveNotes( active  ) );
    }

    const handlePictureClick = () => {
        document.querySelector('#fileSelector').click();
    }

    const  handleFileChange = (e) => {
        console.log(  e.target );
    }


    return (
        <div  className="notes__appbar">
            <span>28 de Agosto del 2021</span>

            <input id="fileSelector" name="file" type="file"  style={{ display: 'none' }} onChange={ handleFileChange } />

            <div> 
            
                <button className="btn"  onClick={ handlePictureClick } >Picture</button>

                <button className="btn" onClick={ handleSave }  >Save</button>

            </div>

        </div>
    )
}
