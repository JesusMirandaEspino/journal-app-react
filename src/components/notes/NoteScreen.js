import React, { useEffect, useRef } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { NotesAppBar } from './NotesAppBar';
import { useForm } from '../../hooks/useForm';
import { activeNote, startDeleting } from '../../actions/notes';

export const NoteScreen = () => {

    const dispatch = useDispatch( );

    const { active: note } = useSelector( state => state.notes );

    const  [ formValues, handleInputChange, reset ] =  useForm( note );

    const { title, body, id } = formValues;

    const activeId = useRef(  note.id );


    useEffect(() => {
        if( note.id !== activeId.current ){
            reset( note );
            activeId.current = note.id;
        }
    }, [ note, reset ]);


    useEffect(() => {
        dispatch( activeNote(  formValues.id, { ...formValues } ) );
    }, [formValues, dispatch]);


    const  handleDelete = () => {
        dispatch( startDeleting( id ) );
    };

    return (
        <div className="note__main-content" >

            <NotesAppBar />

            <div className="notes__content" >
                <label htmlFor="title">Title</label>
                <input  types="text" className="notes__title-input" name="title" placeholder="Some Awasome Title" value={ title } onChange={ handleInputChange } />
                <textarea  className="note__textarea" placeholder="What happened today"  name="body" value={ body } onChange={ handleInputChange } ></textarea>

                { (note.url) &&
                    <div className="notes__image" >
                    <img  src={ formValues.url }  alt="img" />
                </div>
                }

            </div>


        <button className="btn btn-danger"  onClick={  handleDelete } >Delete</button>

        </div>
    )
}
