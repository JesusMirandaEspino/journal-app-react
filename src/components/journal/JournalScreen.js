import React from 'react';
import { Sidebar } from './Sidebar';
// en proceso import { NothingSeleted } from './NothingSeleted';
import { NoteScreen } from '../notes/NoteScreen';

export const JournalScreen = () => {
    return (
        <div className="journal__main-content" >


            <Sidebar />

            <main>

                {/* <NothingSeleted /> */}

                <NoteScreen />

            </main>


        </div>
    )
}
