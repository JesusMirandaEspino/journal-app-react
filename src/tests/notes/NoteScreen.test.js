
import React from 'react';
import { Sidebar } from '../../components/journal/Sidebar';
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import  configureStore  from 'redux-mock-store';
import {configure,  mount } from 'enzyme';
import { Provider } from 'react-redux';
import  thunk  from 'redux-thunk';
import '@testing-library/jest-dom';

import  { activeNote } from '../../actions/notes';
import { NoteScreen } from '../../components/notes/NoteScreen';


jest.mock( '../../actions/notes', () => ({
    activeNote: jest.fn(),
}));




const middleware = [thunk];
const mockstore = configureStore( middleware );

configure({ adapter: new Adapter() });

const initialState = {
    auth: {
        uid: '1',
        name: 'Jesus'
    },
    ui: {
        loading: false,
        msgError: null
    },
    notes: {
        active: {
            id: 123456,
            title: 'Mundo',
            body: 'Hola',
            date: 0
        },
        notes: []
    }
};

let store = mockstore( initialState );
store.dispatch = jest.fn();

        const wrapper = mount(

        <Provider store={ store } >
                <NoteScreen />
        </Provider>

        );

describe('Pruebas en <NoteScreen />', () => {

    test( 'Debe mostrarse Correctamente', () => {

        expect( wrapper ).toMatchSnapshot();

    });


    test( 'Debe de disparar el activeNote', () => {

        wrapper.find('input[name="title"]').simulate( 'change', {
            target: {
                name: 'title',
                value: 'Hola de nuevo'
            }
        });

        expect( activeNote ).toHaveBeenCalled();

    });



});