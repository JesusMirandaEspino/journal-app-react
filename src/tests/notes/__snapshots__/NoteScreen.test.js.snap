// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <NoteScreen /> Debe mostrarse Correctamente 1`] = `
<div
  className="note__main-content"
>
  <div
    className="notes__appbar"
  >
    <span>
      28 de Agosto del 2021
    </span>
    <input
      id="fileSelector"
      name="file"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
        Picture
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="notes__content"
  >
    <label
      htmlFor="title"
    >
      Title
    </label>
    <input
      className="notes__title-input"
      name="title"
      onChange={[Function]}
      placeholder="Some Awasome Title"
      types="text"
      value="Mundo"
    />
    <textarea
      className="note__textarea"
      name="body"
      onChange={[Function]}
      placeholder="What happened today"
      value="Hola"
    />
  </div>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
